src/AspNet.Security.OAuth.SuperOffice/Implementation/DefaultSuperOfficeIdTokenValidator.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AspNet.Security.OAuth.SuperOffice.Implementation
+{
+    internal sealed class DefaultSuperOfficeIdTokenValidator : SuperOfficeIdTokenValidator
```

Comment:
I'm not 100% sure having such a hook is useful. I'd personally remove these two files and move the logic to the handler class itself. Also, please consider adding a `JwtSecurityTokenHandler` property in the options instead of resolving it from DI. You'll even get bonus points if you use `JsonWebTokenHandler` instead of `JwtSecurityTokenHandler` 😄

Reply:
Dumbed it all down and keeping all the validation in the Handler. Thanks. but not taking anymore unnecessary dependencies.

Reply:
I should have been clearer: `JsonWebTokenHandler` is part of `Microsoft.IdentityModel.JsonWebTokens`, which is already referenced by the older/legacy `JwtSecurityTokenHandler` package.

Reply:
Right. Updated.

Reply:
@kevinchalet poke. :-)